cmake/configure/configure_20_trilinos.cmake

```
@@ -272,9 +272,45 @@ macro(feature_trilinos_find_external var)
       else()
         message(
           STATUS
-          "Tpetra was found but is not usable! Disabling Tpetra support."
+          "Tpetra was found but is not usable! Disabling Tpetra support and searching for common errors."
           )
         set(TRILINOS_WITH_TPETRA OFF)
+
+        check_cxx_symbol_exists(HAVE_TPETRA_INT_LONG_LONG "TpetraCore_config.h" DEAL_II_HAVE_TPETRA_INT_LONG_LONG)
+        check_cxx_symbol_exists(HAVE_TPETRA_INT_INT "TpetraCore_config.h" DEAL_II_HAVE_TPETRA_INT_INT)
+
+        if(DEAL_II_HAVE_TPETRA_INT_INT AND DEAL_II_WITH_64BIT_INDICES)
```

This isn't really problem, right? We should rather check that `DEAL_II_HAVE_TPETRA_INT_LONG_LONG` is not defined.
```suggestion
        if(NOT DEAL_II_HAVE_TPETRA_INT_LONG_LONG AND DEAL_II_WITH_64BIT_INDICES)
```

---

Yes and no, since Trilinos does not let you instantiate more than one global ordinate type.
But you are correct that this would be the safer version, same as below.

---

Changed it to check if the needed configuration is missing